ast/src/analyzed/mod.rs

```diff
@@ -1325,6 +1325,8 @@ pub struct PolynomialReference {
     /// The type arguments if the symbol is generic.
     /// Guaranteed to be Some(_) after type checking is completed.
     pub type_args: Option<Vec<Type>>,
+    ///
+    pub resolved_impl: Option<Box<Expression>>,
```

Comment:
If this is an expression, we need to clone it, but we also don't want to have lifetime parameters in this struct. Another option would be to store an index into the implementation vector.